.github/workflows/pr-integration.yml

```diff
@@ -93,3 +93,17 @@ jobs:
       with:
         test-server: workspace-local.json
         test: suites/FullIntegration.json
+
+    - name: "Notify WSM Slack"
+      if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
I have not tested with this condition here (i.e. `if: always() && github.event_name == '**push**' && github.ref == '**refs/heads/main**'`) as that requires…a push to main branch.

I tested against this very PR using `if: always() && github.event_name == '**pull_request**' && github.ref == '**refs/pull/421/merge**'`. I'll ensure to follow up after merging this PR to main to make sure everything looks good.

Reply:
I'm not too well-versed in GHA, but why is the `always()` needed at the beginning of the conditional?

(same question in test.yml)

Reply:
The `always()` tells GHA to run the job no matter what the status (success, failure) is. But yeah, the `always()` function is a bit odd here, especially when then combined with `&&`.

This short, related discussion from [here](https://github.com/DataBiosphere/terra-workspace-manager/pull/378#discussion_r649429130) from earlier may also help.

Reply:
what's the difference between the behavior of these two:
`if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'`
and 
`if: github.event_name == 'push' && github.ref == 'refs/heads/main'`

Reply:
In `if: always() && github.event_name == 'push' && github.ref == 'refs/heads/main'`, results will post to Slack no matter the status of the previous steps (success, failure, cancelled).

In `if: github.event_name == 'push' && github.ref == 'refs/heads/main'`, GitHub will apply the default status check `success()`. Results will post to Slack only when none of the previous steps failed OR was cancelled.

https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions

Reply:
Awesome! Thanks - I think Matt and I asked the same question at the same time - I didn't realize he had left the same comment until I refreshed :)

Reply:
lol no worries